### DISCUSSION TITLE
LYWSD03MMC Bluetooth Sensor Issues

### DISCUSSION THREAD

**Discussion by biohazard827** [theyosh/TerrariumPI#803] [Q&A]
I'm able to get the LYWSD03MMC sensor connected via the RPi Bluetooth, and it works for a little bit, maybe 5 - 10 minutes. After that it stops communicating with the following:

_Invalid reading from sensor LYWSD03MMC bluetooth sensor humidity named..._

I have attached the log where you are able to see it go from working to not working. Are there any know issues with this Bluetooth sensor?

[terrariumpi_logfile.txt](https://github.com/theyosh/TerrariumPI/files/11522517/terrariumpi_logfile.txt)

**Comment by theyosh** [comment]
Hmm, maybe to obvious.. but empty battery? Specially if it worked well for the a long time..

**Reply by biohazard827** [reply]
The devices are new, and only work for 5-10 min at a time before having errors. They will work again for 5-10 minutes if I reboot the RPi. It's almost seems like the Bluetooth device is getting overwhelmed with responses.

**Reply by theyosh** [reply]
Hmm, I just did another look at the log file, and this is what I am seeing.

- The first updates works but takes about 20 seconds to get a readout. That is very long, almost to long, as I update every 30 seconds.
- When you start TP4, there is a line saying: `Bluetooth scanning is not enabled for normal users or there are zero Bluetooth LE devices available.... bluetooth is disabled!`. Which is strange, as you do have bluetooth, and sensors connected. So this message is a bit strange... 
- And after 4 or 5 readouts, I only see errors.

So, I do not know what to think of this. The error about no bluetooth available is something of your PI. What kind of Pi are you using?
And about the long time it needs to read out, I would think of a low battery. But because of the strange message of your Pi about bluetooth scanning, I am doubting....
But how long is the distance between sensor and Pi? And what is in between them? Glas, walls, doors etc?
Are there any other processes running? Is there spikes in the load? 

So, I do not have a single answer... and I do not have the device at home. So am not able to do some debugging... only guessing at this point..:(

**Reply by anhpham140195** [reply]
Hello, I just installed TP4 2 days ago and used LYWSD03MMC as well. In the beginning, I had the same issue, it worked for a little bit, around 5min to 30min I guess, but after that, my RPi could not connect to the device anymore and my RPi is Raspberry Pi zero 2W.
<br>
I have solved this problem, and it has been running smoothly for more than 15 hours now, this is how I did it you can try or someone is having the same problem as me, hope this helps:
- Update to the latest firmware version of LYWSD03MMC
- Increase __ENGINE_LOOP_TIMEOUT in terrariumEngine.py to 60s (from my side sometime it take more than 30s so that why I increase it)

Good luck!
